migrations/privateProgramInProjects/updatePrivateProgramInProject.js

```diff
@@ -0,0 +1,152 @@
+            projectIds = await chunkOfProjectDocument[pointerToProject].map(
+                projectDoc => {
+                  return projectDoc._id;
+                }
+            );
+
+
+            // get project documents from projects collection in Array
+            let projectDocuments = await db.collection('projects').find({
+                _id: { $in :projectIds  }
+            }).project({}).toArray();
+            //iterate project documents one by one
+            for(let counter = 0; counter < projectDocuments.length; counter++) {
+
+
+                if(projectDocuments[counter].hasOwnProperty("solutionId")){
+                    // find solution document form solution collection
+                    let solutionDocument = await db.collection('solutions').find({
+                        _id: projectDocuments[counter].solutionId
+                    }).project({}).toArray({})
+                    //find program document form program collection
+                    if(solutionDocument[0].hasOwnProperty("parentSolutionId")){
+                       
+                        // find parent solution document in same collection
+                        let parentSolutionDocument = await db.collection('solutions').find({
+                            _id: solutionDocument[0].parentSolutionId}).project({}).toArray({});
+                        //varibale to update project document
+                        let updateProjectDocument = {
+                            "$set" : {}
+                        };
+                        updateProjectDocument["$set"]["solutionId"] = parentSolutionDocument[0]._id
+                        updateProjectDocument["$set"]["isAPrivateProgram"] = parentSolutionDocument[0].isAPrivateProgram
+                        updateProjectDocument["$set"]["solutionInformation"] = {
+                            name: parentSolutionDocument[0].name,   
+                            description: parentSolutionDocument[0].description,
+                            externalId: parentSolutionDocument[0].externalId,
+                            _id: parentSolutionDocument[0]._id,
+                        }
+                        updateProjectDocument["$set"]["solutionExternalId"] = parentSolutionDocument[0].externalId,
+                        updateProjectDocument["$set"]["programId"] = parentSolutionDocument[0].programId,
+                        updateProjectDocument["$set"]["programExternalId"] =  parentSolutionDocument[0].programExternalId
+                        updateProjectDocument["$set"]["programInformation"] = {
+                            _id : parentSolutionDocument[0].programId,
+                            name : parentSolutionDocument[0].programName,
+                            externalId : parentSolutionDocument[0].programExternalId,
+                            description : parentSolutionDocument[0].programDescription,
```

Comment:
@ankitpws is this key present in the solution?

Reply:
@aks30 In parent Solution document all this data is avilable